core/coreapi/interface/interface.go

```diff
@@ -178,27 +181,90 @@ type KeyAPI interface {
 }
 
-// type ObjectAPI interface {
-// 	New() (cid.Cid, Object)
-// 	Get(string) (Object, error)
-// 	Links(string) ([]*Link, error)
-// 	Data(string) (Reader, error)
-// 	Stat(string) (ObjectStat, error)
-// 	Put(Object) (cid.Cid, error)
-// 	SetData(string, Reader) (cid.Cid, error)
-// 	AppendData(string, Data) (cid.Cid, error)
-// 	AddLink(string, string, string) (cid.Cid, error)
-// 	RmLink(string, string) (cid.Cid, error)
-// }
-
-// type ObjectStat struct {
-// 	Cid            cid.Cid
-// 	NumLinks       int
-// 	BlockSize      int
-// 	LinksSize      int
-// 	DataSize       int
-// 	CumulativeSize int
-// }
+// ObjectAPI specifies the interface to MerkleDAG and contains useful utilities
+// for manipulating MerkleDAG data structures.
+type ObjectAPI interface {
+	// New creates new, empty (by default) dag-node.
+	New(context.Context, ...options.ObjectNewOption) (Node, error)
+
+	// WithType is an option for New which allows to change the type of created
+	// dag node.
+	//
+	// Supported types:
+	// * 'empty' - Empty node
+	// * 'unixfs-dir' - Empty UnixFS directory
+	WithType(string) options.ObjectNewOption
+
+	// Put imports the data into merkledag
+	Put(context.Context, io.Reader, ...options.ObjectPutOption) (Path, error)
+
+	// WithInputEnc is an option for Put which specifies the input encoding of the
+	// data. Default is "json".
+	//
+	// Supported encodings:
+	// * "protobuf"
+	// * "json"
+	WithInputEnc(e string) options.ObjectPutOption
+
+	// WithDataType specifies the encoding of data field when using Josn or XML
+	// input encoding.
+	//
+	// Supported types:
+	// * "text" (default)
+	// * "base64"
+	WithDataType(t string) options.ObjectPutOption
+
+	// Get returns the node for the path
+	Get(context.Context, Path) (Node, error)
+
+	// Data returns reader for data of the node
+	Data(context.Context, Path) (io.Reader, error)
+
+	// Links returns lint or links the node contains
+	Links(context.Context, Path) ([]*Link, error)
+
+	// Stat returns information about the node
+	Stat(context.Context, Path) (*ObjectStat, error)
+
+	// AddLink adds a link under the specified path. child path can point to a
+	// subdirectory within the patent which must be present (can be overridden
+	// with WithCreate option).
+	AddLink(ctx context.Context, base Path, name string, child Path, opts ...options.ObjectAddLinkOption) (Path, error)
+
+	// WithCreate is an option for AddLink which specifies whether create required
+	// directories for the child
+	WithCreate(create bool) options.ObjectAddLinkOption
+
+	// RmLink removes a link from the node
+	RmLink(ctx context.Context, base Path, link string) (Path, error)
+
+	// AppendData appends data to the node
+	AppendData(context.Context, Path, io.Reader) (Path, error)
+
+	// SetData sets the data contained in the node
+	SetData(context.Context, Path, io.Reader) (Path, error)
+}
+
+// ObjectStat provides information about dag nodes
+type ObjectStat struct {
```

Comment:
Should I wrap this into an interface for consistency?

Reply:
I think it's fine - given that the object API is quasi-frozen (it's the old merkledag/dag-pb structure), it's very unlikely this struct will ever have any functions attached to it.